src/core/server/elasticsearch/client/client_config.ts

```diff
@@ -59,6 +59,10 @@ export function parseClientOptions(
     // do not make assumption on user-supplied data content
     // fixes https://github.com/elastic/kibana/issues/101944
     disablePrototypePoisoningProtection: true,
+    agent: {
+      maxSockets: Infinity,
+      keepAlive: config.keepAlive,
```

Comment:
Let's configure `keepAlive` fallback explicitly to avoid a situation when Kibana depends on ES client defaults
```js
keepAlive: config.keepAlive ?? true,
```